src/config.rs

```diff
@@ -0,0 +1,211 @@
+use std::str::FromStr;
+
+use strum::{EnumProperty, IntoEnumIterator};
+use strum_macros::{AsRefStr, Display, EnumIter, EnumProperty, EnumString};
+
+use crate::constants::DC_VERSION_STR;
+use crate::context::Context;
+use crate::dc_job::*;
+use crate::dc_stock::*;
+use crate::dc_tools::*;
+use crate::x::*;
+
+/// The available configuration keys.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Display, EnumString, AsRefStr, EnumIter, EnumProperty,
+)]
+#[strum(serialize_all = "snake_case")]
+pub enum Config {
+    Addr,
+    MailServer,
+    MailUser,
+    MailPw,
+    MailPort,
+    SendServer,
+    SendUser,
+    SendPw,
+    SendPort,
+    ServerFlags,
+    #[strum(props(default = "INBOX"))]
+    ImapFolder,
+    Displayname,
+    Selfstatus,
+    Selfavatar,
+    #[strum(props(default = "1"))]
+    E2eeEnabled,
+    #[strum(props(default = "1"))]
+    MdnsEnabled,
+    InboxWatch,
+    #[strum(props(default = "1"))]
+    SentboxWatch,
+    #[strum(props(default = "1"))]
+    MvboxWatch,
+    #[strum(props(default = "1"))]
+    MvboxMove,
+    #[strum(props(default = "0"))]
+    ShowEmails,
+    SaveMimeHeaders,
+    ConfiguredAddr,
+    ConfiguredMailServer,
+    ConfiguredMailUser,
+    ConfiguredMailPw,
+    ConfiguredMailPort,
+    ConfiguredSendServer,
+    ConfiguredSendUser,
+    ConfiguredSendPw,
+    ConfiguredSendPort,
+    ConfiguredServerFlags,
+    Configured,
+}
+
+// deprecated
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Display, EnumString, AsRefStr, EnumIter)]
+pub enum SysConfig {
+    #[strum(serialize = "sys.version")]
+    Version,
+    #[strum(serialize = "sys.msgsize_max_recommended")]
+    MsgsizeMaxRecommended,
+    #[strum(serialize = "sys.config_keys")]
+    ConfigKeys,
+}
+
+/// Get a configuration key.
+/// Returns "" when the key is invalid, or no default was found.
+pub fn get(context: &Context, key: impl AsRef<str>) -> String {
```

Comment:
Also should we not make `key` and enum and make the from string stuff only a C-API compatibility?